vega_sim/scenario/ideal_market_maker_v2/agents.py

```diff
@@ -235,35 +264,40 @@ def step(self, vega_state: VegaState):
 
         buy_order, sell_order = None, None
 
-        for order in (
-            vega_state.market_state[self.market_id]
-            .orders[self.vega.wallet.public_key(self.wallet_name)]
-            .values()
-        ):
-            if order.side == vega_protos.SIDE_BUY:
-                buy_order = order
-            else:
-                sell_order = order
-
-        # self.vega.submit_simple_liquidity(
-        #     wallet_name=self.wallet_name,
-        #     market_id=self.market_id,
-        #     commitment_amount=self.commitment_amount,
-        #     fee=0.002,
-        #     reference_buy="PEGGED_REFERENCE_MID",
-        #     reference_sell="PEGGED_REFERENCE_MID",
-        #     delta_buy=self.bid_depth,
-        #     delta_sell=self.ask_depth,
-        # )
-
-        self._place_orders(
-            buy_offset=self.bid_depth,
-            sell_offset=self.ask_depth,
-            volume=1,
-            buy_order=buy_order,
-            sell_order=sell_order,
+        try:
+            for order in (
+                vega_state.market_state[self.market_id]
+                .orders[self.vega.wallet.public_key(self.wallet_name)]
+                .values()
+            ):
+                if order.side == vega_protos.SIDE_BUY:
+                    buy_order = order
+                else:
+                    sell_order = order
+            
+        except Exception as error:
+            print(error)
+            pass
+
+        self.vega.submit_simple_liquidity(
+            wallet_name=self.wallet_name,
+            market_id=self.market_id,
+            commitment_amount=self.commitment_amount,
+            fee=0.002,
+            reference_buy="PEGGED_REFERENCE_BEST_BID",
+            reference_sell="PEGGED_REFERENCE_BEST_ASK",
+            delta_buy=self.bid_depth - self.spread,
```

Comment:
This should be `- spread/2` I think?

Reply:
Spread would (should) be difference between best bid and best ask, not mid-price to best X.

Reply:
Another idea: how about not to set the self.spread for market_maker, instead to let market_maker call the information of current LOB to know the current best bid/ask and spread of the market?

Reply:
I'm not sure the market maker actually even uses the spread it is given anymore looking at it

Reply:
But yeah I think the spread shape in BackgroundMarket should be fixed too 👍